application/src/main/kotlin/net/corda/v5/application/crypto/DefaultSignatureSpecService.kt

```diff
@@ -0,0 +1,26 @@
+package net.corda.v5.application.crypto
+
+import net.corda.v5.crypto.SignatureSpec
+import java.security.PublicKey
+
+interface DefaultSignatureSpecService {
+    /**
+     * Work out a signature and hash algorithm to use for a given public key, given current security policies.
+     *
+     * @param publicKey the public key to be used for signing
+     *
+     * @return An appropriate [SignatureSpec], or null if nothing is available for the key type.
+     */
```

Comment:
Needs @Suspendable

Reply:
Okay. It may be that we don't need to suspend the fiber to implement the API, but it may also be that we do since we want to go the crypto service for configuration.